common-http/src/main/java/io/cdap/common/http/HttpResponse.java

```diff
@@ -86,12 +90,44 @@ public byte[] getResponseBody() {
     return responseBody;
   }
 
+  /**
+   * Decodes and return the response body based on the content encoding.
+   */
+  public byte[] getUncompressedResponseBody() {
+    String encoding = headers.entries().stream()
+      .filter(e -> HttpHeaders.CONTENT_ENCODING.equalsIgnoreCase(e.getKey()))
+      .map(Map.Entry::getValue)
+      .findFirst()
+      .orElse(null);
+
+    if (encoding == null) {
+      return responseBody;
+    }
+
+    try {
+      if ("gzip".equalsIgnoreCase(encoding)) {
+        try (InputStream is = new GZIPInputStream(new ByteArrayInputStream(responseBody))) {
+          return ByteStreams.toByteArray(is);
+        }
+      }
+      if ("deflate".equalsIgnoreCase(encoding)) {
+        try (InputStream is = new DeflaterInputStream(new ByteArrayInputStream(responseBody))) {
+          return ByteStreams.toByteArray(is);
+        }
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException(String.format("Failed to decompress %s encoded response body", encoding), e);
+    }
+
+    throw new IllegalArgumentException("Unsupported content encoding " + encoding);
```

Comment:
Use `IllegalStateException` as it's not an argument

Reply:
fixed